### PR TITLE
[dca2] add dev mode field for dev

### PR DESCRIPTION
**Pull request by kbearXD**

### CODE REVIEW

File: pkg/strategy/dca2/dev_mode.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package dca2
+
+// DevMode, if Enabled is true. it means it will check the running field
+type DevMode struct {
+	On            bool `json:"on"`
```
Comment (c9s) [inline pkg/strategy/dca2/dev_mode.go:5]: use `enabled`

File: pkg/strategy/dca2/dev_mode.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+	IsNewStrategy bool `json:"isNewStrategy"`
```
Comment (c9s) [inline pkg/strategy/dca2/dev_mode.go:6]: how about `newAccount` ?

File: pkg/strategy/dca2/strategy.go [hunk 3]
```diff
@@ -66,6 +66,9 @@ type Strategy struct {
 	// UseCancelAllOrdersApiWhenClose uses a different API to cancel all the orders on the market when closing a grid
 	UseCancelAllOrdersApiWhenClose bool `json:"useCancelAllOrdersApiWhenClose"`
 
+	// dev mode
+	DevMode DevMode `json:"devMode"`
```
Comment (c9s) [inline pkg/strategy/dca2/strategy.go:70]: would be better to use `DevMode *DevMode`

File: pkg/strategy/dca2/open_position.go [hunk 4]
```diff
@@ -32,6 +33,21 @@ func (s *Strategy) placeOpenPositionOrders(ctx context.Context) error {
 
 	s.debugOrders(createdOrders)
 
+	if s.DevMode.On && s.DevMode.IsNewStrategy {
```
Comment (c9s) [inline pkg/strategy/dca2/open_position.go:36]: and check s.DevMode != nil here

File: pkg/strategy/dca2/dev_mode.go (hunk 2 above)
Comment (kbearXD) [reply pkg/strategy/dca2/dev_mode.go:6]: Actually, it's not new account. This field is used to make dca2 strategy not recover with long time ago trades.

File: pkg/strategy/dca2/strategy.go (hunk 3 above)
Comment (kbearXD) [reply pkg/strategy/dca2/strategy.go:70]: I think if we use pointer here, we can remove Enabled field.

**Review by c9s — APPROVED** [review]

**Pull request merged**